pika/adapters/asyncio_connection.py

```diff
@@ -0,0 +1,168 @@
+        :param loop: Asyncio Loop
+
+        """
+        self.loop = loop
+
+    def add_timeout(self, deadline, callback_method):
+        """Add the callback_method to the EventLoop timer to fire after deadline
+        seconds. Returns a Handle to the timeout.
+
+        :param int deadline: The number of seconds to wait to call callback
+        :param method callback_method: The callback method
+        :rtype: asyncio.Handle
+
+        """
+        return self.loop.call_later(deadline, callback_method)
+
+    @staticmethod
+    def remove_timeout(handle):
+        """
+        Cancel asyncio.Handle
+
+        :type handle: asyncio.Handle
+        :rtype: bool
+        """
+        return handle.cancel()
+
+    def add_handler(self, fd, cb, event_state):
+        """ Registers the given handler to receive the given events for ``fd``.
+
+        The ``fd`` argument is an integer file descriptor.
+
+        The ``event_state`` argument is a bitwise or of the constants
+        ``base_connection.BaseConnection.READ``, ``base_connection.BaseConnection.WRITE``,
+        and ``base_connection.BaseConnection.ERROR``.
+
+        """
+
+        if event_state & base_connection.BaseConnection.READ:
+            self.loop.add_reader(
+                fd,
+                partial(
+                    cb,
+                    fd=fd,
+                    events=base_connection.BaseConnection.READ
+                )
+            )
+
+        if event_state & base_connection.BaseConnection.WRITE:
+            self.loop.add_writer(
+                fd,
+                partial(
+                    cb,
+                    fd=fd,
+                    events=base_connection.BaseConnection.WRITE
+                )
+            )
+
+    def remove_handler(self, fd):
+        """ Stop listening for events on ``fd``. """
+
+        self.loop.remove_reader(fd)
+        self.loop.remove_writer(fd)
+
+    def update_handler(self, fd, event_state):
+        raise NotImplementedError()
+
+    def start(self):
+        """ Start Event Loop """
+        if self.loop.is_running():
+            return
+
+        self.loop.run_forever()
+
+    def stop(self):
+        """ Stop Event Loop """
+        if self.loop.is_closed():
+            return
+
+        self.loop.stop()
+
+
+class AsyncioConnection(base_connection.BaseConnection):
+    """ The AsyncioConnection runs on the Asyncio EventLoop.
+
+    :param pika.connection.Parameters parameters: Connection parameters
+    :param on_open_callback: The method to call when the connection is open
+    :type on_open_callback: method
+    :param on_open_error_callback: Method to call if the connection cant be opened
+    :type on_open_error_callback: method
+    :param asyncio.AbstractEventLoop loop: By default asyncio.get_event_loop()
+
+    """
+    def __init__(self, parameters=None, on_open_callback=None,
+                 on_open_error_callback=None,
+                 on_close_callback=None, loop=None):
```

Comment:
@mosquito, please model the constructor args on TornadoConnection/SelectConnection for consistency. This should also facilitate easy integration into the async adapter acceptance test framework. See 

* https://github.com/pika/pika/blob/43936088e1183ec5902f2ba2992c902e87f5460b/pika/adapters/tornado_connection.py#L30-L35
* https://github.com/pika/pika/blob/43936088e1183ec5902f2ba2992c902e87f5460b/pika/adapters/select_connection.py#L68-L73